### ISSUE TITLE
How to control the padding between charts?

### ISSUE THREAD

**Issue by Aix3D** [juj/RectangleBinPack#16]
As the title says. : )

**Comment by juj** [comment]
Yeah, sorry, this fell through my attention.

Currently the code allows nothing much better to pack bigger rectangle.

Though if we think about padding, one often wants to have a "collapse padding" feature. I.e. say we are packing rectangles with a padding of `p` units. In regular manner we'd pack a rectangle of `(W+2*p)x(H+2*p)` to increase the size by `p` in all four directions. This ends up each content rectangle to be at least `2*p` units apart from each other, even if one wanted content only to be `p` units apart.

To "collapse padding", you could try finding free spots for a `(W+2*p)x(H+2*p)` rectangle, but after such a spot has been found, only record a used rectangle of `WxH` in the center of the resulting space. That way when the adjacent rectangles are packed, their padding will be allowed to overlap with the existing rectangles.

**Comment by Aix3D** [comment]
Hi, this fell through my attention too these days.
Now I have injected RectangleBinPack into our project, a lightmap baking tool. I have processed the padding around the chart and it works well. Wonderful!
I considered whether take account padding into the interface is a good design when I injected it into our project days before, and as a result, I think including padding into interface may cause some confusion. Consider the width and height of a chart only is a clean design, I can control the padding myself, and it works as what I want.
Thanks for your replay.
